File: matiasmasso/Mat.NET/DTO/DTOCertificatIrpf.cs
using System;

namespace DTO
{
    public class DTOCertificatIrpf : DTOBaseGuid
    {
        public string filename { get; set; }
        public string Nif { get; set; }
        public string Nom { get; set; }
        public DTODocFile DocFile { get; set; }
        public DTOContact Contact { get; set; }
        public int Year { get; set; }
        public int Period { get; set; }


        public DTOCertificatIrpf() : base()
        {
        }

        public DTOCertificatIrpf(Guid oGuid) : base(oGuid)
        {
        }


        public static DTOCertificatIrpf Factory(string sFilename)
        {
            DTOCertificatIrpf retval = new DTOCertificatIrpf();
            retval.filename = sFilename;
            return retval;
        }


        public string FullPeriod()
        {
            var retval = Year.ToString();
            if (Period == 0)
                retval = string.Format("{0:0000}", Year);
            else
                retval = string.Format("{0:0000}-{1:00}", Year, Period);
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOCertificatIrpf.vb
Public Class DTOCertificatIrpf
    Inherits DTOBaseGuid
    Property filename As String
    Property Nif As String
    Property Nom As String
    Property DocFile As DTODocFile
    Property Contact As DTOContact
    Property Year As Integer
    Property Period As Integer


    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub


    Shared Function Factory(sFilename As String) As DTOCertificatIrpf
        Dim retval As New DTOCertificatIrpf
        retval.filename = sFilename
        Return retval
    End Function

    Shared Function Factory(exs As List(Of Exception), year As Integer, filenames As List(Of String)) As List(Of DTOCertificatIrpf)
        Dim retval As New List(Of DTOCertificatIrpf)
        For Each sfilename In filenames
            Dim item = DTOCertificatIrpf.Factory(sfilename)
            Dim src = iTextPdfHelper.readText(sfilename, exs)
            If exs.Count = 0 Then
                Dim lines = src.Split(vbLf).ToList
                Dim keyLine = lines.FirstOrDefault(Function(x) x.Contains("Datos del Perceptor"))
                If keyLine > "" Then
                    Dim idx = lines.IndexOf(keyLine)
                    If lines.Count > idx + 3 Then
                        Dim sLine = lines(idx + 2)
                        Dim iPos = sLine.IndexOf(" ")
                        item.Nif = sLine.Substring(0, iPos)
                        If item.Nif.StartsWith("0") Then item.Nif = item.Nif.Substring(1)
                        item.Nom = sLine.Substring(iPos + 1)
                        item.Year = year
                        retval.Add(item)
                    End If
                End If
            Else
                exs.Add(New Exception("error al llegir '" & sfilename & "'"))
            End If
        Next
        Return retval
    End Function

    Public Function FullPeriod() As String
        Dim retval = _Year.ToString
        If _Period = 0 Then
            retval = String.Format("{0:0000}", _Year)
        Else
            retval = String.Format("{0:0000}-{1:00}", _Year, _Period)
        End If
        Return retval
    End Function
End Class
